prog/rand.go

```diff
@@ -138,15 +138,33 @@ func (r *randGen) randPageCount() (n uint64) {
 }
 
 // Change a flag value or generate a new one.
-func (r *randGen) flags(vv []uint64, bitmask bool, oldVal uint64) (v uint64) {
+func (r *randGen) flags(vv []uint64, size uint64, bitmask bool, oldVal uint64) (v uint64) {
 	v = oldVal
+	if r.nOutOf(1, 5) {
+		// Ignore the old value sometimes.
+		v = 0
+	}
 	switch {
+	case (bitmask && r.nOutOf(1, 2)):
+		// Try flipping random bits.
+		for stop := false; !stop; stop = r.oneOf(3) {
+			bit := r.rand(int(size) * 8)
```

Comment:
The current mutation code looks wrong, but this looks suboptimal. Consider we have 3 flags and uint64. 19 out of 20 flips are bad. We are lucky if kernel will just ignore them, but it may also return EINVAL. So I think we need to use value as the primary source of relevant bits.
It can make sense to flip completely random bits, but with the same low probability we do "v = r.rand64()". To make it perfect we could flip bits adjacent to flags values  with higher probability, e.g. if flags contain first 4 bits them maybe 5-th and 6-th are also relevant (just because we forgot to add them), but 27-th and 28-th are most likely not.

Reply:
Do I understand correctly, that you're suggesting to remove flipping random bits and only flip bits that come from `vv` here?

Reply:
Yes, at least most of the time.

Reply:
Done, PTAL.